flow/activities/flowable.go

```diff
@@ -636,8 +653,15 @@ func (a *FlowableActivity) RecordSlotSizes(ctx context.Context) error {
 		return err
 	}
 
-	logger := activity.GetLogger(ctx)
+	recordedPeers := make(map[string]bool)
+
 	for _, config := range configs {
+		peerName := config.Source.Name
+		if recordedPeers[peerName] {
+			continue
+		}
+		recordedPeers[peerName] = true
+
 		func() {
```

Comment:
Not seeing why this block needs to be wrapped in closure

Reply:
it was like that earlier too, i think for the defer

Reply:
Yes